### ISSUE TITLE
Dynamically added picker causes resize JS error

### ISSUE THREAD

**Issue by pbrf01** [snapappointments/bootstrap-select#2105]
Dynamically added and then hidden select pickers cause a JS error when you resize the browser.  To reproduce:
https://plnkr.co/edit/PnylbTpbC6AZhZIh0OXi?p=preview

1. Click 'Add Picker' button to dynamically create a new bootstrap-select picker.
2. Select an option.
3. Click 'Disable Picker' button to hide the selectpicker (destroy causes the same behavior).
4. Resize the browser.  You'll get a JS error about the resizing (lots of errors actually): 

```
"bootstrap-select.js:820 Uncaught TypeError: Cannot read property '0' of undefined
    at scroll (bootstrap-select.js:820)
    at bootstrap-select.js:918
    at dispatch (jquery.min.js:3)
    at q.handle (jquery.min.js:3)"
```

I believe it's only when the select picker is dynamically added and then removed.
    
Happens in: IE 11, Firefox 52, Chrome 66 and 68, Safari 10. Both Windows and Mac OS.
Environment: Bootstrap 4.1.x, bootstrap-select 1.13.2

This may or may not relate to Issue #2034.

**Comment by WizardCM** [comment]
I've come across a similar issue, and I believe it's part of a larger problem.

There are two issues here:

1. The big one - a lot of the event functions (in my case getSelectPosition) either don't use the target in the event, or **always expect the element to exist and be visible**, which isn't always the case. When it's not the case, it immediately errors instead of cancelling.
2. The reason the issue still occurs on destroy, is because **the destroy function doesn't actually remove any event hooks**. This one is a big big big memory leak in my opinion, and a problem in general because every time .selectpicker() is run on an element, even after a 'destroy', the event hooks for scroll and resize are stacked. And stacked. And stacked. It's really, really bad.

**Comment by caseyjhol** [comment]
Released in [v1.13.4](https://github.com/snapappointments/bootstrap-select/releases/tag/v1.13.4)!
